modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIFilterRelevantObjectEntryModelListenerTest.java

```diff
@@ -87,4 +111,245 @@ public void test() throws Exception {
 			jsonObject.get("title"));
 	}
 
+	@Test
+	public void testGetEndpointsWithApiEndpointToAPIFiltersAsNestedFields()
+		throws Exception {
+
+		_addAPIApplication(
+			_objectDefinitionJSONObject1.getString("externalReferenceCode"));
+
+		_addAPIFilter("name eq 'myTest'");
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"items",
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"apiEndpointToAPIFilters",
+						JSONUtil.putAll(
+							JSONUtil.put("oDataFilter", "name eq 'myTest'"))))
+			).put(
+				"lastPage", 1
+			).put(
+				"page", 1
+			).put(
+				"pageSize", 20
+			).put(
+				"totalCount", 1
+			).toString(),
+			HTTPTestUtil.invokeToJSONObject(
+				null,
+				"headless-builder/endpoints" +
+					"?nestedFields=apiEndpointToAPIFilters",
+				Http.Method.GET
+			).toString(),
+			JSONCompareMode.LENIENT);
+	}
+
+	@Test
+	public void testGetWithAPIFilterAsNestedFieldsWithNestedFieldsDepth()
+		throws Exception {
+
+		_addAPIApplication(
+			_objectDefinitionJSONObject1.getString("externalReferenceCode"));
+
+		_addAPIFilter("name eq 'myTest' or name eq '1@ ab9'");
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"items",
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"apiApplicationToAPIEndpoints",
+						JSONUtil.putAll(
+							JSONUtil.put(
+								"apiEndpointToAPIFilters",
+								JSONUtil.putAll(
+									JSONUtil.put(
+										"oDataFilter",
+										"name eq 'myTest' or name eq '1@ ab" +
+											"9'"))))))
+			).put(
+				"lastPage", 1
+			).put(
+				"page", 1
+			).put(
+				"pageSize", 20
+			).put(
+				"totalCount", 1
+			).toString(),
+			HTTPTestUtil.invokeToJSONObject(
+				null,
+				"headless-builder/applications" +
+					"?nestedFields=apiApplicationToAPIEndpoints," +
+						"apiEndpointToAPIFilters&nestedFieldsDepth=2",
+				Http.Method.GET
+			).toString(),
+			JSONCompareMode.LENIENT);
+	}
+
+	private void _addAPIApplication(
```

Comment:
Could we use these new methods in the existing test? So we create the application in the same way in all tests